File: Count_Complete_Tree_Nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int countNodes(TreeNode root) {
        if(root==null)
            return 0;
     
        int left = getLeftHeight(root)+1;    
        int right = getRightHeight(root)+1;
     
        if(left==right){
            return (2<<(left-1))-1;
        }else{
            return countNodes(root.left)+countNodes(root.right)+1;
        }
    }
     
    public int getLeftHeight(TreeNode n){
        if(n==null) return 0;
     
        int height=0;
        while(n.left!=null){
            height++;
            n = n.left;
        }
        return height;
    }
     
    public int getRightHeight(TreeNode n){
        if(n==null) return 0;
     
        int height=0;
        while(n.right!=null){
            height++;
            n = n.right;
        }
        return height;
    }
}

File: Swift/count-complete-tree-nodes.swift
/**
 * https://leetcode.com/problems/count-complete-tree-nodes/
 * 
 * 
 */ 
// Date: Sun Oct 24 19:49:18 PDT 2021
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *     }
 * }
 */
class Solution {
    func countNodes(_ root: TreeNode?) -> Int {
        if root == nil { return 0 }
        var leftNode = root
        var rightNode = root
        var left = 0
        var right = 0
        while leftNode != nil {
            left += 1
            leftNode = leftNode?.left
        }
        while rightNode != nil {
            right += 1
            rightNode = rightNode?.right
        }
        if left == right { return Int(pow(2, Double(left))) - 1 }
        return 1 + self.countNodes(root?.left) + self.countNodes(root?.right)
    }
}